tools/markdown-to-html/docs-marked-renderer.spec.ts

```diff
@@ -0,0 +1,89 @@
+import {DocsMarkdownRenderer} from './docs-marked-renderer';
+
+describe('DocsMarkdownRenderer', () => {
+  let renderer: DocsMarkdownRenderer;
+  beforeEach(() => {
+    renderer = new DocsMarkdownRenderer();
+  });
+
+  it('generates regular headings for h1 and h2', () => {
+    expect(renderer.heading('a', 1, 'ignored')).toEqual('<h1>a</h1>');
+    expect(renderer.heading('b', 2, 'ignored')).toEqual('<h2>b</h2>');
+  });
+
+  it('creates header link for h3 and h4 headings', () => {
+    const heading3 = renderer.heading('heading text', 3, 'link-id');
+    expect(heading3).toEqual(`
+        <h3 id="link-id" class="docs-header-link">
```

Comment:
Does this break if we change the number of leading whitespace on each line? If so maybe you can do something like this:
```
expect(heading3.replace(/\s/g, '')).toEqual(`
  ...
`.replace(/\s/g, ''))
```

Reply:
Yea, changing the leading whitespace would have broken the tests. It was actually kind of annoying to get the whitespace correct in the first place :P 
Added helper methods to ignore leading whitespace.